clientapi/auth/storage/accounts/storage.go

```diff
@@ -28,6 +30,69 @@ import (
 	_ "github.com/lib/pq"
 )
 
+type AccountDatabase interface {
+	GetAccountByPassword(
+		ctx context.Context, localpart, plaintextPassword string,
+	) (*authtypes.Account, error)
+	GetProfileByLocalpart(
+		ctx context.Context, localpart string,
+	) (*authtypes.Profile, error)
```

Comment:
Please document the exact contract you wish to expose here. E.g if there is no profile for a given localpart, do we:
 - return an error?
 - return a nil profile and nil error?

Can the storage impl do string coercion e.g case insensitive matching? I appreciate that there may be no 'right' answer here, in which case please document what your code does under these circumstances so we can check the behaviour.